logic.js

```diff
@@ -0,0 +1,195 @@
+//  a module to serve as our database and useful methods
+
+
+const Tweeter = function () {
+    const _posts = [  // posts and comments data base
+        {
+            id: "p1",
+            text: "First post!",
+            comments: [
+                { id: "c1", text: "First comment on first post!" },
+                { id: "c2", text: "Second comment on first post!!" },
+                { id: "c3", text: "Third comment on first post!!!" }
+            ]
+        },
+        {
+            id: "p2",
+            text: "Aw man, I wanted to be first",
+            comments: [
+                { id: "c4", text: "Don't wory second poster, you'll be first one day." },
+                { id: "c5", text: "Yeah, believe in yourself!" },
+                { id: "c6", text: "Haha second place what a joke." }
+            ]
+        }
+    ]
+
+
+// available methods of Tweeter 
+
+
+let postsCounter = _posts.length+1  // counter for number of posts already stored
+ 
+// add a new post function by accepting a new string from user
+
+const addPost = function (string){
+    const add =  function () {
+    postsCounter += 1;
+      }
+    _posts.push({   id:"p"+postsCounter,
+                        text: string ,
+                        comments:[] })
+                        add() 
```

Comment:
Why invoking a function inside another function, and not just to write the code inside the `addPost`?